v4/source/system.yaml

```
@@ -1095,3 +1095,48 @@
                 format: binary
         "404":
           $ref: "#/components/responses/NotFound"
+  /integrity:
+    post:
+      tags:
+        - system
+      summary: Perform a database integrity check
+      description: >
+        Performs a database integrity check.
+
+
+        __Note__: This check may temporarily harm system performance.
+
+
+        __Minimum server version__: 5.28.0
+
+
+        __Local mode only__: This endpoint is only available through [local mode](https://docs.mattermost.com/administration/mmctl-cli-tool.html#local-mode).
+      responses:
+        "200":
+          description: Integrity check successful
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/IntegrityCheckResult"
+      x-code-samples:
+        - lang: Go
+          source: |
+            import (
+                "net"
+                "net/http"
+
+                "github.com/mattermost/mattermost-server/v5/model"
+            )
+
+            tr := &http.Transport{
+                Dial: func(network, addr string) (net.Conn, error) {
+                    return net.Dial("unix", socketPath)
+                },
+            }
+
+            Client := model.NewAPIv4Client("http://_")
+            Client.HttpClient = &http.Client{Transport: tr}
```

nit: recently [we merged the `NewAPIv4SocketClient` function](https://github.com/mattermost/mattermost-server/blob/7602dc0b19ebc3e4aaad3a5a9cd27dc897ab516e/model/client4.go#L122), we can use it here:

```suggestion
            Client := model.NewAPIv4SocketClient(socketPath)
```